### PR TITLE
Match update and event flag fix

### PR DESCRIPTION
**Pull request by YannickSoehngen**
Fixed GoodEventFlag, increased granularity from counter to Get4 level, added pulser Flag (previous part of GoodEventFlag) and implementation of Single-Sided-Matches

### CODE REVIEW

File: StRoot/StEvent/StETofHeader.h
Comment (klendathu2k) [inline StRoot/StEvent/StETofHeader.h]: Addition of mHasPulserVec on line 114 requires incrementing the argument to ClassDef.

File: StRoot/StMuDSTMaker/COMMON/StMuETofHeader.h
Comment (klendathu2k) [inline StRoot/StMuDSTMaker/COMMON/StMuETofHeader.h]: Additional data field (line 121) requires incrementing the ClassDef.

File: StRoot/StPicoEvent/StPicoEvent.h
Comment (klendathu2k) [inline StRoot/StPicoEvent/StPicoEvent.h:620]: ClassDef needs to be incremented.

**Comment by plexoos** [comment]
There was a small conflict with the main branch. I resolved it

File: StRoot/StETofCalibMaker/StETofCalibMaker.cxx
Comment (plexoos) [inline StRoot/StETofCalibMaker/StETofCalibMaker.cxx]:
```suggestion
		      hasPulsersVec.push_back(mNPulsersCounter[iCounter] == 2);    
```

File: StRoot/StETofCalibMaker/StETofCalibMaker.cxx
Comment (plexoos) [inline StRoot/StETofCalibMaker/StETofCalibMaker.cxx]:
```suggestion
		    goodEventFlagVec.push_back(!etofHeader->missMatchFlagVec().at(iGet4)); 
```

File: StRoot/StETofCalibMaker/StETofCalibMaker.cxx
Comment (plexoos) [inline StRoot/StETofCalibMaker/StETofCalibMaker.cxx]:
```suggestion
		    goodEventFlagVec.push_back( !etofHeader->missMatchFlagVec().at(iGet4) );
```

File: StRoot/StETofCalibMaker/StETofCalibMaker.cxx
Comment (plexoos) [inline StRoot/StETofCalibMaker/StETofCalibMaker.cxx]:
```suggestion
		      hasPulsersVec.push_back(mNPulsersCounter[iCounter] == 2);   
```

**Review by plexoos — COMMENTED** [review]
Please consider reducing the number of if-statements

**Comment by plexoos** [comment]
There are also errors from the compiler. Please address them

File: StRoot/StETofCalibMaker/StETofCalibMaker.cxx
Comment (plexoos) [inline StRoot/StETofCalibMaker/StETofCalibMaker.cxx]: This should be re-written in modern C++ without using an explicit loop. The both source and destination vectors appear to contain boolean values, so using std::copy with back_inserter should work.

**Review by klendathu2k — APPROVED** [review]
StEvent etc... looks good now.

**Review by plexoos — APPROVED** [review]

**Comment by genevb** [comment]
Are we waiting on anything here? A review from @PhilippWeidenkaff ? Or is this ready to merge?

**Review by fgeurts — APPROVED** [review]

**Comment by genevb** [comment]
I understand that this is the only PR on which we are critically waiting before tagging SL24a. I'll ask again: are we waiting for anything else on this PR before merging?

**Comment by genevb** [comment]
Nightly tests look like they went OK. @plexoos , would you please tag SL24a here and `star-mcgen` when you get the chance? Thanks in advance.

**Comment by plexoos** [comment]
Done. Branch SL24a and tag SL24a_0 should be available in the both repositories now

**Pull request merged**
